thorlcr/activities/lookupjoin/thlookupjoincommon.hpp

```diff
@@ -0,0 +1,41 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#ifndef THLOOKUPSLAVECOMMON_HPP
+#define THLOOKUPSLAVECOMMON_HPP
+
+#include "eclhelper.hpp"
+
+#include "thgraph.hpp"
+
+enum SmartJoinState:byte { sjs_undefined, sjs_globallookup, sjs_locallookup, sjs_std };
```

Comment:
This will need to move when the enum gets translated to text, but that can happen later.